src/utilities/network/lookupApi.js

```diff
@@ -19,68 +22,219 @@ const SEPARATORS = {
   POLICY_NUMBER: '-',
 };
 
-const getQueryString = ({
+export const createPatientRecord = response => {
```

Comment:
Yeah, agree. The main issue is that the sync creators expect the incoming data to be in "external" format, so I think the tidiest way of doing it might be to make a separate creator helper which both use (and maybe the incoming sync logic maps the exterrnal fields to internal before calling it?).

Reply:
hmm I'm a bit slow - I think you're saying `createRecord` expects "external" format (snake_case) - I could be not understanding you though. It's actually just for general use rather than for sync - https://github.com/openmsupply/mobile/blob/d00f1274640307c64fadd8a18df877ba20375954/src/database/utilities/createRecord.js#L113-L142 or for a prescriber: https://github.com/openmsupply/mobile/blob/d00f1274640307c64fadd8a18df877ba20375954/src/database/utilities/createRecord.js#L89-L101

It's super close to being able to just pass your mapped object and removing your methods

Reply:
I do agree with a common method for sync/not sync though.. some sort of `deserialize`/`serialize` methods maybe - would be nice for each data type... possibly a `create` static method in each class wou;d be a cleaner approach also.. would at least save us from the soon to be 4d `createRecord` 3000 line file..

Reply:
Yeah, I think we're on the same page here (I wasn't distinguishing properly between `createRecord` and the creation logic in `incomingSyncUtils`).